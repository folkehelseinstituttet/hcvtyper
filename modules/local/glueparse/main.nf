process GLUEPARSE {

    label 'process_single'

    // Environment with R tidyverse and jsonlite packages from the conda-forge channel. Created using seqera containers.
    // URLs:
    // Docker image: https://wave.seqera.io/view/builds/bd-fcb57a72bf127ac8_1?_gl=1*ipusgc*_gcl_au*MTkxMjgxNTMwMi4xNzUzNzczOTQz
    // Singularity image: https://wave.seqera.io/view/builds/bd-739b63e9fb54f431_1?_gl=1*420hyj*_gcl_au*MTkxMjgxNTMwMi4xNzUzNzczOTQz
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/a2/a29ad4a97f1b927d4c34550beb44729a7ce4c95be62ede9a96153939c0878b79/data':
        'community.wave.seqera.io/library/r-jsonlite_r-tidyverse:2.0.0--fcb57a72bf127ac8' }"

    input:
    path '*'

    output:
    path("*.tsv")      , emit: GLUE_summary
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    GLUE_json_parser.R \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        tidyverse: \$(Rscript -e "library(tidyverse); cat(as.character(packageVersion('tidyverse')))")
        seqinr: \$(Rscript -e "library(seqinr); cat(as.character(packageVersion('seqinr')))")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    // TODO nf-core: A stub section should mimic the execution of the original module as best as possible
    //               Have a look at the following examples:
    //               Simple example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bcftools/annotate/main.nf#L47-L63
    //               Complex example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bedtools/split/main.nf#L38-L54
    // TODO nf-core: If the module doesn't use arguments ($args), you SHOULD remove:
    //               - The definition of args `def args = task.ext.args ?: ''` above.
    //               - The use of the variable in the script `echo $args ` below.
    """
    echo $args

    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        glueparse: \$(glueparse --version)
    END_VERSIONS
    """
}
