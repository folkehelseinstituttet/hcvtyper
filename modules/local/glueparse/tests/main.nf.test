nextflow_process {

    name "Test Process GLUEPARSE"
    script "../main.nf"
    process "GLUEPARSE"

    tag "modules"
    tag "glueparse"

    // Real run (uses your JSON in the module tests folder)
    test("glueparse: real JSON") {
        when {
            process {
                """
                // Input signature: path '*'
                input[0] = file("${projectDir}/modules/local/glueparse/tests/Test_2.3a_D17763_major.major.nodup.json", checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot([
                    GLUE_summary: process.out.GLUE_summary,
                    versions    : process.out.versions
                ]).match() }
            )
        }
    }

    // Fast stub run
    test("glueparse: stub") {
        options "-stub-run"
        when {
            process {
                """
                input[0] = file("${projectDir}/modules/local/glueparse/tests/Test_2.3a_D17763_major.major.nodup.json", checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot([
                    GLUE_summary: process.out.GLUE_summary,
                    versions    : process.out.versions
                ]).match() }
            )
        }
    }

}
