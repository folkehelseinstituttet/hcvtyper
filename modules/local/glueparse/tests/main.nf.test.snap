{
    "glueparse: real JSON": {
        "content": [
            {
                "GLUE_summary": [
                    [
                        "GLUE_collected_report_major.tsv:md5,16ba6ce94fd04c726829f1752cce6f94",
                        "GLUE_collected_report_minor.tsv:md5,dbac9e27dc30c54ab781aaca85e3d887"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2693e68a7ccc95c54fdd7b1ed501c03f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-06T15:45:18.948138739"
    },
    "glueparse: stub": {
        "content": [
            {
                "GLUE_summary": [
                    "GLUE_summary.tsv:md5,dbac9e27dc30c54ab781aaca85e3d887"
                ],
                "versions": [
                    "versions.yml:md5,c44e5b8d1e7f0f28758617183c153555"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-06T15:45:29.590592831"
    }
}